### ISSUE TITLE
`next` and `previous` freeze when given invalid input

### ISSUE THREAD

**Issue by daviskirk** [sdispater/pendulum#121]
The `next` and `previous` methods freeze when given an invalid input.  Using python 3.6.1 and python 3.5.3:

```python
dt = pendulum.create(2012, 1, 31, 12, 56, 0)
dt.previous(pendulum.SUNDAY)
```

returns `<Pendulum [2012-01-29T00:00:00+00:00]>` as expected but 

```python
dt = pendulum.create(2012, 1, 31, 12, 56, 0)
dt.previous(7)
```

just freezes the interpreter.

**Comment by sdispater** [comment]
It's fixed on the `master` branch (c2df79d80775b1f55f4741da810703aebd1c3367).

It will land in the next bugfix release.

**Comment by sdispater** [comment]
It's fixed as of version `1.2.2`!

**Comment by daviskirk** [comment]
Awesome! Thanks for the great library!
